cleanup fabric api violations, remove persistent fabric flags (3 of n)

SeanNijjar opened the pull request:
remove persistent fabric flags because fabric is always persistent. These are dead-code. Removing them partly cleans up the API violation but mainly simplifies remaining cleanup work.

https://github.com/tenstorrent/tt-metal/issues/20621

### Checklist
- [x] [All post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml) CI: https://github.com/tenstorrent/tt-metal/actions/runs/15565278381
- [x] [Blackhole Post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml) CI with demo tests passes (if applicable): https://github.com/tenstorrent/tt-metal/actions/runs/15565283894
- [x] microbench: https://github.com/tenstorrent/tt-metal/actions/runs/15571940106
- [x] T3K choose your own adventure: https://github.com/tenstorrent/tt-metal/actions/runs/15565271575

bbradelTT reviewed (approved):
Is the umd bump intentional?

SeanNijjar committed:
cleanup fabric api violations (3 of n)

remove persistent fabric flags because fabric is always persistent. These are dead-code. Removing them partly cleans up the API violation but mainly simplifies remaining cleanup work.

fix regressions

fix regressions

updates after rebase

SeanNijjar committed:
fix test regression

SeanNijjar committed:
update targets

omilyutin-tt reviewed (approved):
Looks great, very nice clean up! 🥇
